### ISSUE TITLE
BUG. Sitemap.xml don't show page if it is hidden in default locale

### ISSUE THREAD

**Issue by uniun** [tractorcow/silverstripe-fluent#110]
Sitemap.xml don't show page if it is hidden in default locale with FluentFilteredExtension (but that page is visible on any other locale).

**Comment by RoyalPulp** [comment]
Hi! This problem still exists!
Is there any workaround?
Thanks for any idea!

**Comment by RoyalPulp** [comment]
...or would it be possible to generate seperate sitemap Files for each locale?

**Comment by tractorcow** [comment]
Yes you should have a separate sitemap.xml for each locale. I haven't tested it for a while though.

Which version of fluent are you using?

**Comment by RoyalPulp** [comment]
I am running  SilverStripe 3.6.5 with fluent 3.8.1

**Comment by RoyalPulp** [comment]
How do I access locale-based sitemap files? The code looks like sitemap.xml is generated from the base-locale

**Comment by tractorcow** [comment]
Oh sorry, that would be the case if you had domain based localisation.

E.g. www.mysite.ca/sitemap.xml vs www.mysite.com.au/sitemap.xml

With urlsegment based localisation, yes you are correct it would be missing those default locale records.

**Comment by tractorcow** [comment]
It's possible we'll need different sitemap.xml for multi-domain and single-domain modes.

**Comment by g4b0** [comment]
I'm running a single-domain site with the following modules installed:

```json
        "tractorcow/silverstripe-fluent": "^4.1",
        "silverstripe/googlesitemaps": "^2.1"
```

Is it possible to have two different sitemaps for each locale?
Thanks

**Comment by DarrenInwood** [comment]
If it helps, I solved this via:
https://pastebin.com/G3aWrT6q

**Comment by RoyalPulp** [comment]
okay, thank you, any idea how to adapt this für SilverStripe 3.7.1 ?

**Comment by g4b0** [comment]
Thankyou @DarrenInwood, nice workaround. Unfortunately it doesn't work with DataObjects, I will have a look at it in the next few days.
